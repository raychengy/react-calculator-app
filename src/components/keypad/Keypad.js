import React from 'react';
import PropTypes from 'prop-types';
import KeypadRow from './KeypadRow';

const Keypad = props => (
    <section className="keypad">
        {props.btnLayout.map(
            (rowLayout, i) => (
                <KeypadRow key={`keypad__row__${i}`} rowLayout={rowLayout} onClick={props.onClick} />
            ))}
    </section>
)

Keypad.propTypes = {
    btnLayout: PropTypes.array.isRequired,
    onClick: PropTypes.func.isRequired,
}

export default Keypad;