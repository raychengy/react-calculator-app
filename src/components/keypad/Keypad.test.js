import React from 'react';
import renderer from 'react-test-renderer';

import Keypad from './Keypad';

const btnLayout = [
    [{ displayName: "C", value: "C", classNames: ["operator"] }, { displayName: <span>&larr;</span>, value: "bksp", classNames: ["operator"] }, { displayName: "%", value: "%", classNames: ["operator"] }, { displayName: <span>&divide;</span>, value: "/", classNames: ["operator"] }],
    [{ displayName: "9", value: "9", }, { displayName: "8", value: "8", }, { displayName: "7", value: "7", }, { displayName: <span>&times;</span>, value: "*", classNames: ["operator"] }],
    [{ displayName: "4", value: "4", }, { displayName: "5", value: "5", }, { displayName: "6", value: "6", }, { displayName: "+", value: "+", classNames: ["operator"] }],
    [{ displayName: "1", value: "1", }, { displayName: "2", value: "2", }, { displayName: "3", value: "3", }, { displayName: "-", value: "-", classNames: ["operator"] }],
    [{ displayName: "0", value: "0", }, { displayName: ".", value: ".", }, { displayName: "=", value: "=", classNames: ["btn--large"] }],
];

const onClick = jest.fn();

describe('<Keypad />', () => {
    describe('Using props', () => {
        it('renders without crashing', () => {
            const keypad = renderer.create(<Keypad btnLayout={btnLayout} onClick={onClick} />);
            expect(keypad.toJSON()).toMatchSnapshot();
        });
    })
});
