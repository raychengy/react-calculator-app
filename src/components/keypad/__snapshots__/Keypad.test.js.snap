// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keypad /> Using props renders without crashing 1`] = `
<section
  className="keypad"
>
  <div
    className="keypad__row"
  >
    <button
      className="btn operator"
      onClick={[Function]}
    >
      C
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      <span>
        ←
      </span>
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      %
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      <span>
        ÷
      </span>
    </button>
  </div>
  <div
    className="keypad__row"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      9
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      7
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      <span>
        ×
      </span>
    </button>
  </div>
  <div
    className="keypad__row"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="keypad__row"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div
    className="keypad__row"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      .
    </button>
    <button
      className="btn btn--large"
      onClick={[Function]}
    >
      =
    </button>
  </div>
</section>
`;
